ml/rl/workflow/identify_types_flow.py

```diff
@@ -0,0 +1,109 @@
+    )
+    return spark
+
+
+def normalization_helper(**kwargs):
+    norm_params = {
+        "max_unique_enum_values": kwargs["max_unique_enum_values"],
+        "quantile_size": kwargs["quantile_size"],
+        "quantile_k2_threshold": kwargs["quantile_k2_threshold"],
+        "skip_box_cox": kwargs.get("skip_box_cox", False),
+        "skip_quantiles": kwargs.get("skip_quantiles", False),
+        "feature_overrides": kwargs["feature_overrides"],
+    }
+    whitelist_features = set(kwargs.get("whitelist_features", None) or [])
+    whitelist_features_assert = kwargs.get("whitelist_features_assert", True)
+
+    def validate_whitelist_features(
+        params: Dict[int, normalization.NormalizationParameters],
+    ) -> None:
+        if not whitelist_features:
+            return
+        whitelist_feature_set = {int(fid) for fid in whitelist_features}
+        available_features = set(params.keys())
+        assert whitelist_feature_set == available_features, (
+            "Could not identify preprocessing type for these features: {}; "
+            "extra features: {}".format(
+                whitelist_feature_set - available_features,
+                available_features - whitelist_feature_set,
+            )
+        )
+
+    def process(df) -> Dict[int, normalization.NormalizationParameters]:
+        params = dict()
+        for row in df.collect():
+            norm_metdata = normalization.get_feature_norm_metadata(
+                row["feature_name"], row["feature_values"], norm_params
+            )
+            if norm_metdata is not None and (
+                not whitelist_features or row["feature_name"] in whitelist_features
+            ):
+                params[row["feature_name"]] = norm_metdata
+
+        if whitelist_features_assert:
+            validate_whitelist_features(params)
+        return params
+
+    return process
+
+
+def identify_normalization_parameters(
+    table_spec: TableSpec, column_name: str, preprocessing_options: PreprocessingOptions
+) -> Dict[int, normalization.NormalizationParameters]:
+    """ Get normalization parametters """
+
+    spark = get_spark_session(table_spec.warehouse_dir)
+    df = spark.sql(f"SELECT * FROM {table_spec.table}")
+    df = create_normalization_spec_spark(df, column_name, preprocessing_options)
+
+    normalization_processor = normalization_helper(
+        max_unique_enum_values=preprocessing_options.max_unique_enum_values,
+        quantile_size=preprocessing_options.quantile_size,
+        quantile_k2_threshold=preprocessing_options.quantile_k2_threshold,
+        skip_box_cox=preprocessing_options.skip_box_cox,
+        skip_quantiles=preprocessing_options.skip_quantiles,
+        feature_overrides=preprocessing_options.feature_overrides,
+        whitelist_features=preprocessing_options.whitelist_features,
+        whitelist_features_assert=preprocessing_options.assert_whitelist_feature_coverage,
+    )
+
+    return normalization_processor(df)
```

Comment:
I think we should shutdown spark session before leaving this function, e.g., with try-finally